### PR TITLE
Manage orders and transactions

### PR DESCRIPTION
**Pull request by cnivedit**
- Create commands
- Edit to use Item and ItemMap
- Create methods to list order and transaction
- Add exception messages
- Update parser

### CODE REVIEW

File: src/main/java/seedu/pill/util/Order.java [hunk 1]
```diff
@@ -55,19 +55,35 @@ public Order(OrderType type, String notes) {
...
+    /**
+     * Creates a new Order with the specified type and notes.
+     * The order is automatically assigned a unique identifier and initialized
+     * with a PENDING status and the current timestamp.
+     *
+     * @param type  - The type of order (PURCHASE or DISPENSE)
+     * @param notes - Additional information or comments about the order
+     */
+    public Order(OrderType type, ItemMap itemsToOrder, String notes) {
+        this.id = UUID.randomUUID();
```
Comment (yijiano) [inline src/main/java/seedu/pill/util/Order.java:71]: Great use of UUID to label our orders.

File: src/main/java/seedu/pill/util/Parser.java [hunk 2]
```diff
@@ -117,6 +150,202 @@ public void parseCommand(String input) {
...
+     *                       <ul>
+     *                           <li>{@code PillException} with {@code ExceptionMessages.TOO_MANY_ARGUMENTS} if more
+     *                           than one argument is provided.</li>
+     *                           <li>{@code PillException} with {@code ExceptionMessages.INVALID_FULFILL_COMMAND} if no
+     *                           argument is provided.</li>
+     *                           <li>{@code PillException} with {@code ExceptionMessages.INVALID_INDEX} if the argument
+     *                           cannot be parsed as an integer.</li>
+     *                           <li>{@code PillException} with {@code ExceptionMessages.INVALID_ORDER} if the specified
+     *                           order index is out of bounds.</li>
+     *                       </ul>
```
Comment (yijiano) [inline src/main/java/seedu/pill/util/Parser.java:171]: The JavaDocs here seems a bit odd here. Perhaps we can shorten this, and move the existing details in full to the DG.

File: src/main/java/seedu/pill/util/Parser.java [hunk 3]
```diff
@@ -117,6 +150,202 @@ public void parseCommand(String input) {
...
+
+                    if (j != itemArguments.length - 1) {
+                        throw new PillException(ExceptionMessages.INVALID_ITEM_FORMAT);
+                    }
+                }
+
+                if (isANumber(currentArgument)) {
+                    quantityIndex = j;
+                }
+            }
```
Comment (yijiano) [inline src/main/java/seedu/pill/util/Parser.java:318]: Perhaps there can be a way to abstract this part out to a separate helper function?

**Review by yijiano — APPROVED** [review]
LGTM, thank you for your hard work.

File: src/main/java/seedu/pill/util/Parser.java (hunk 2 above)
Comment (cnivedit) [reply src/main/java/seedu/pill/util/Parser.java]: Edited in [79e3c2b](https://github.com/AY2425S1-CS2113-W14-4/tp/pull/142/commits/79e3c2bb8551cf52774c68cddb0a1e055f37a211)

File: src/main/java/seedu/pill/util/Parser.java (hunk 3 above)
Comment (cnivedit) [reply src/main/java/seedu/pill/util/Parser.java]: Extracted out in 68759bf

**Pull request merged**